src/Generator/DiffGenerator.php

```diff
@@ -70,7 +70,7 @@ static function ($assetName) use ($filterExpression) {
             ! method_exists($this->schemaManager, 'getSchemaSearchPaths')
             && $this->platform->supportsSchemas()
         ) {
-            $defaultNamespace = $toSchema->getName();
+            $defaultNamespace = $toSchema->getNamespaceName() ?? '';
```

Comment:
This doesn't look right. The schema _is_ the namespace for its tables. The schema itself cannot be associated with a namespace (only tables, views and sequences can). Currently it's possible to call on any `getNamespaceName()` (which is a design flaw) but in DBAL 5, it will be possible only with the objects mentioned earlier.

The only situation in which currently a schema can be associated with a namespace is when its name contains a dot and the name isn't quoted (which is also wrong).